File: kotlin/src/main/kotlin/Subscription.kt
typealias Action = (Subscription, MassiveDataChangedEvent) -> Unit

class Subscription(private val connectionSession: IMassiveSession) {

    var publishingEnabled: Boolean = false
    var publishingInterval: Int = 0

    fun create() {
        TODO("can't be called from a unit test")
    }

    fun registerCallback(subscriptionDataChange: Action) {
        TODO("can't be called from a unit test")
    }

    fun delete() {
        TODO("can't be called from a unit test")
    }
}

File: typescript/src/Subscription.ts
import { IMassiveSession } from './IMassiveSession';
import { MassiveDataChangedEvent } from './MassiveDataChangedEvent';

type SubscriptionDataChange = (subscription: Subscription, event: MassiveDataChangedEvent) => void;

export class Subscription {
  publishingEnabled!: boolean;
  publishingInterval!: number;

  constructor(private readonly _connectionSession: IMassiveSession) {
  }

  create() {
    throw new Error("can't be called from a unit test");
  }

  registerCallback(_subscriptionDataChange: SubscriptionDataChange) {
    throw new Error("can't be called from a unit test");
  }

  delete() {
    throw new Error("can't be called from a unit test");
  }
}
